pyfar/dsp/regularization.py

```diff
@@ -0,0 +1,313 @@
+"""
+Documentation for regularization class.
+"""
+
+import pyfar as pf
+from pyfar.dsp.dsp import _cross_fade
+import numpy as np
+
+class RegularizedSpectrumInversion():
+    r"""Class for frequency dependent regularization and inversion.
+
+    Regularization is used in inverse filtering methods to limit the gain
+    applied by an inverse filter. This, for example, is nescessary to
+    avoid extreme amplification caused by notches and peaks in the original
+    signal's spectrum or by the limit of a system's bandwidth.
+
+    The frequency dependent regularization :math:`\epsilon(f)` can be defined
+    using one of the class methods.
+    The regularization is then applied when calculating the inverse of a
+    signal as [1]_:
+
+    .. math::
+
+        S^{-1}(f) = \frac{S^*(f)}{S^*(f)S(f) + \beta * |\epsilon(f)|^2} * D(f)
+
+    with :math:`S(f)` being the input signal's Fourier transform,
+    :math:`S^*(f)` the complex conjugate, and :math:`\beta` a parameter to
+    control the influence of the regularization function on the inversion.
+    :math:`D(f)` denotes an optional arbitrary target function.
```

Comment:
I would skip applying the target function here. it is making it more complex then it is, and the Class is called Inversion.